compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1030,6 +1030,8 @@ pub enum ProjectionElem<V, T> {
     /// Like an explicit cast from an opaque type to a concrete type, but without
     /// requiring an intermediate variable.
     OpaqueCast(T),
+
+    Subtype(T),
```

Comment:
Please add a doc comment explaining the behavior and constraints of this projection.

Reply:
I think we should add something like this here:

- this projection does not impact runtime behavior except for potentially changing some type metadata of the interpreter or codegen backend.